### PR TITLE
Changed the module loader to conform to intended behavior

### PR DESCRIPTION
**Pull request by chuck211991**
I also ended up adding in a library; this is just a start though, much more to do.

### CODE REVIEW

File: module-loader.js [hunk 1]
```diff
@@ -92,13 +182,81 @@ ModuleLoader.prototype.open = function(blob, options) {
...
+
+ModuleLoader.prototype.getApplicationURL = function(module, api_url) {
+    //var module_name = this.slugify(module);
+    // API doesn't support slugged module names yet
+    var module_name = module;
+    return api_url + module_name + '/?format=json';
+}
+
+ModuleLoader.prototype.slugify = function(text) {
+  // From https://gist.github.com/mathewbyrne/1280286
```
Comment (ibara) [inline module-loader.js]: We should keep the entire original comment since the code hasn't changed.

**Comment by mskmoorthy** [comment]
LGTM

**Comment by hughed2** [comment]
This looks fine. A lot of it is stuff that I wanted to change and was just waiting to hear back from you after you added in your dependency tree code, so you've saved me some trouble. :)

**Comment by hughed2** [comment]
I'm writing a comment on the community site pull request, wait for that and check it out before doing anything else here.

**Pull request merged**